### ISSUE TITLE
[BR]: fail2ban-regex fails with multiple ignoreregex patterns

### ISSUE THREAD

**Issue by brevilo** [fail2ban/fail2ban#3895]
### Environment:

- Fail2Ban version  :  0.11.2-2
- OS, including release name/version : Debian 11.11 (bullseye)
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
Whenever I use _multiple_ patterns for `ignoreregex` a subsequent test throws an exception:
```
failregex = PATTERN

ignoreregex = PATTERN1
              PATTERN2
```
Notes on tests done:
- The `ignoreregex` patterns themselves don't matter. I _tested both_ alone/independently and _they work_ just fine
- Pattern anchoring (`xxx` vs. `^xxx$`) doesn't change anything
- The order of the patterns doesn't change anything
- The indentation used for the second line (TABs vs. SPACEs) doesn't change anything

#### Steps to reproduce
```
# fail2ban-regex --print-all-ignored access_log apache-http-dos apache-http-dos
```
#### Expected behavior
Test doesn't throw an exception

#### Observed behavior
```
Running tests
=============

Use   failregex filter file : apache-http-dos, basedir: /etc/fail2ban
Use ignoreregex line : apache-http-dos
Use         log file : access_log
Use         encoding : UTF-8

Traceback (most recent call last):
  File "/usr/bin/fail2ban-regex", line 34, in <module>
    exec_command_line()
  File "/usr/lib/python3/dist-packages/fail2ban/client/fail2banregex.py", line 836, in exec_command_line
    if not fail2banRegex.start(args):
  File "/usr/lib/python3/dist-packages/fail2ban/client/fail2banregex.py", line 776, in start
    self.process(test_lines)
  File "/usr/lib/python3/dist-packages/fail2ban/client/fail2banregex.py", line 584, in process
    line_datetimestripped, ret, is_ignored = self.testRegex(line)
  File "/usr/lib/python3/dist-packages/fail2ban/client/fail2banregex.py", line 456, in testRegex
    found = self._filter.processLine(line, date)
  File "/usr/lib/python3/dist-packages/fail2ban/server/filter.py", line 674, in processLine
    return self.findFailure(tupleLine, date, noDate=noDate)
  File "/usr/lib/python3/dist-packages/fail2ban/server/filter.py", line 865, in findFailure
    if self.__ignoreRegex and self._ignoreLine(buf, orgBuffer, failRegex) is not None:
  File "/usr/lib/python3/dist-packages/fail2ban/server/filter.py", line 734, in _ignoreLine
    if self.onIgnoreRegex: self.onIgnoreRegex(fnd, ignoreRegex)
  File "/usr/lib/python3/dist-packages/fail2ban/client/fail2banregex.py", line 446, in _onIgnoreRegex
    self._ignoreregex[idx].inc()
IndexError: list index out of range
```

**Comment by sebres** [comment]
Thx for the report!..
Indeed, I can confirm it, but it is rather a bit "wrong" usage - if you use the filter name, there is no need (and even seems to be bad) to supply it twice, the second time it is only supposed for the `ignoreregex` itself, not for the filter name.
So this would be correct test:
```diff
- fail2ban-regex --print-all-ignored access_log apache-http-dos apache-http-dos
+ fail2ban-regex --print-all-ignored access_log apache-http-dos
```

What happens in the case where you supply it twice:
Firstly it'd add to the filter instance N `ignoreregex` read from config `filter.d/apache-http-dos.conf` and then overwrite them in the fail2banregex with single value what is `apache-http-dos` (so it shall use this one as `ignoreregex`), but still use N pieces in the filter. So then it would generate IndexError if 2nd to Nth regex will match.
Anyway I'll fix it now.

**Comment by sebres** [comment]
Fixed in 4151eeccfe179e94e23add7ce503bd26ce864e2b, tests pass, thus merge. 
Thx again.

**Comment by brevilo** [comment]
Nice one. Thanks

Some background: the usage doc suggested that using the filter filename was also an option (the one you just removed). Elsewhere I read that you could either use an absolute filename or just the filter's basename, so I chose that for brevity. IIRC, there are examples out there that describe that specific way of calling `fail2ban-regex` to pass the `ignoreregex` argument. But I agree, it always felt weird, but who am I, right...? 😉
